execution/executor-types/src/in_memory_state_calculator.rs

```diff
@@ -245,39 +266,43 @@ impl InMemoryStateCalculator {
 pub fn process_write_set(
     transaction: Option<&Transaction>,
     state_cache: &mut HashMap<StateKey, Option<StateValue>>,
+    usage: &mut StateStorageUsage,
     write_set: WriteSet,
 ) -> Result<HashMap<StateKey, Option<StateValue>>> {
     // Find all keys this transaction touches while processing each write op.
     write_set
         .into_iter()
         .map(|(state_key, write_op)| {
-            process_state_key_write_op(transaction, state_cache, state_key, write_op)
+            process_state_key_write_op(transaction, state_cache, usage, state_key, write_op)
         })
         .collect::<Result<_>>()
 }
 
 fn process_state_key_write_op(
     transaction: Option<&Transaction>,
     state_cache: &mut HashMap<StateKey, Option<StateValue>>,
+    usage: &mut StateStorageUsage,
     state_key: StateKey,
     write_op: WriteOp,
 ) -> Result<(StateKey, Option<StateValue>)> {
+    let key_size = state_key.size();
     let state_value = match write_op {
         WriteOp::Modification(new_value) | WriteOp::Creation(new_value) => {
-            Some(StateValue::from(new_value))
+            let value = StateValue::from(new_value);
+            usage.items += 1;
```

Comment:
does the item differentiate Creation vs Modification?

Reply:
Doesn't matter for calculating total size. We can charge differently before losing that information (when still in the adaptor).

Reply:
Yes, this information will likely only be used for gas metering in the short term. However other components can follow up should they find the need